pinot-plugins/pinot-input-format/pinot-protobuf/src/main/java/org/apache/pinot/plugin/inputformat/protobuf/ProtoBufRecordExtractor.java

```diff
@@ -62,7 +62,7 @@ private Object getFieldValue(Descriptors.FieldDescriptor fieldDescriptor, Messag
     // Note w.r.t proto3 - If a field is not declared with optional keyword, there's no way to distinguish
     // if its explicitly set to a proto default or not been set at all i.e hasField() returns false
     // and we would use null.
-    if (fieldDescriptor.isRepeated() || !fieldDescriptor.isOptional() || message.hasField(fieldDescriptor)) {
+    if (fieldDescriptor.isRepeated() || !fieldDescriptor.hasOptionalKeyword() || message.hasField(fieldDescriptor)) {
```

Comment:
This is exactly what my PR changed - from hasOptionalKeyword to isOptional

https://github.com/apache/pinot/pull/11682

API is deprecated
https://github.com/protocolbuffers/protobuf/commit/d6157f7c7ec78a6942cfd6d96d2e0b40e007ae0d

Reply:
It is deprecated in latest versions of protobuf, but:
- It is not deprecated in the version we use
- There is no alternative to this method in the latest protobuf version.

Theoretically we should is `isPresent`, but it is not honoring its javadoc. You can verify that by changing this to `isPresent` and see that it returns false for the test added here.

Reply:
It seems isPresent is not bug, but I was unable to correctly configure protobuf model programmatically.